apache/superset
# fix(Chart Annotation modal): Table and Superset annotation options will paginate, exceeding previous max limit 100

### SUMMARY

Replace the SelectControl component with an AsyncSelect component when choosing the source of the annotation. This addresses the response limiting table options to 100 max. There are relevant changes to the process of fetching and where the fetch is being made in the file.

### BEFORE/AFTER SCREENSHOTS OR ANIMATED GIF

BEFORE:
![limited-charts](https://github.com/apache/superset/assets/92495987/f5aeaebc-8690-4fc0-9d55-9372c7869407)
AFTER:
![all-charts](https://github.com/apache/superset/assets/92495987/75d5bca9-984f-4b2e-92fa-3b3f7093131f)


### TESTING INSTRUCTIONS

- Create 100+ annotations or Tables. Saving dashboards as duplicates and selecting the duplicate charts box will speed this process up.
- Charts must be **created by you** (annotation options are filtered by `'chart_owned_created_favored_by_me'`)
- Create a new chart
- Click the "Add Annotation" button
- Choose an Annotation type requiring a query, e.g. "Event"
- Scroll through the options to verify display of all 100+ charts

### ADDITIONAL INFORMATION


- [x] Has associated issue: Fixes #25421
